src/plugins/dashboard/server/plugin.ts

```diff
@@ -83,9 +83,9 @@ export class DashboardPlugin
     this.logger.debug('dashboard: Started');
 
     if (plugins.taskManager) {
-      scheduleDashboardTelemetry(this.logger, plugins.taskManager);
-
-      plugins.taskManager.runNow(TASK_ID);
+      scheduleDashboardTelemetry(this.logger, plugins.taskManager)!.then(() => {
+        plugins.taskManager.runSoon(TASK_ID);
+      });
     }
```

Comment:
As we switch to pull model with runSoon, we have to wait the task to be created before trying to run it.